exonum-java-binding/core/rust/src/storage/proof_map_index.rs

```diff
@@ -611,26 +638,39 @@ pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ProofMapInde
     handle::drop_handle::<ProofMapIndexValues<Value>>(&env, iter_handle);
 }
 
-fn convert_to_key(env: &JNIEnv, array: jbyteArray) -> JniResult<Key> {
-    let bytes = env.convert_byte_array(array)?;
-    assert_eq!(PROOF_MAP_KEY_SIZE, bytes.len());
+// Converts array of Java bytes arrays to the vector of keys.
+fn convert_to_keys(env: &JNIEnv, array: jobjectArray) -> JniResult<Vec<Key>> {
+    let num_elements = env.get_array_length(array)?;
+    let mut keys = vec![];
+    for i in 0..num_elements {
+        let byte_array: jbyteArray = env.get_object_array_element(array, i)?.into_inner();
```

Comment:
[Reference management in loops](https://docs.rs/jni/0.14.0/jni/struct.JavaVM.html#local-reference-management)

Reply:
Is that correct that AutoLocals don't work well with any type wrappers (= they erase the type destroying any extra type-safety that these wrappers originally provide)?

Reply:
Here no wrappers are used at all, but sys types.

Reply:
Actually yes, if one works with that wrapper only via `auto_local.as_obj()` (which one is supposed to do since the moment the original object was "wrapped" with `AutoLocal`) then there is no way to see what is exactly behind the returned `JObject`.